test/tests/functional/pbs_root_owned_script.py

```diff
@@ -156,3 +154,29 @@ def test_root_owned_job_array_executable(self):
         ja_comment = "Job Array Held, too many failed attempts to run subjob"
         self.server.expect(JOB, {ATTR_state: "H", ATTR_comment: (MATCH_RE,
                            ja_comment)}, attrop=PTL_AND, id=jid)
+
+    def test_root_owned_job_pbs_attach(self):
+        """
+        submit a job as root and test pbs_attach feature.
+        """
```

Comment:
Do we need something similar to what is present in test_multi_sched.py like the following:

        if os.getuid() != 0 or sys.platform in ('cygwin', 'win32'):
            self.skipTest("Test need to run as root")

Reply:
@mike0042 : This testsuite, pbs_root_owned_script.py is requred to be run as root, and in its setUp(), we already have this:
    def setUp(self):
        """
        Set up the parameters required for Test_RootOwnedScript
        """
        if os.getuid() != 0:
            self.skipTest("Test need to run as root")
        TestFunctional.setUp(self)
---
By the way, I pushed another commit as the ptl file had a pep8 error.

Reply:
The implementation in mom_comm.c is dependent on whether the system is running Linux. Should we also be checking the sys.platform where the test is running to make sure it's not Windows?

Reply:
Oh sure. Ok, I'll add the not Windows test and push another commit.

Reply:
Thanks @bayucan!

Reply:
PTL doesn't run on windows.  I don't think we need to worry about it.  I've actually been seeing people pull that check recently.

Reply:
@bhroam : It might be ported and run on Windows in the future.

Reply:
But aren't there plans for it to run on Windows eventually?

Reply:
I think there are plans to run PTL on Windows soon, @kjakkali can you please confirm?

Reply:
@agrawalravi90 . One of the team is working on Windows-PTL work.